### PR TITLE
Add solution to forms task

### PR DESCRIPTION
**Pull request by DianaGeorgievaa**

### CODE REVIEW

File: Creating_form.html [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        <option value="4">ЯКН</option>
+      </select>
+
+      <label for="course-credits">Кредити</label>
+      <input type="number "id="course-credits" name="credits">  
+
+       <input type="submit" value="Submit">
+    </form>
+  </body>
+</html>
```
Comment (martinely95) [inline Creating_form.html]: Nice job for separating the files by their responsibility!

File: Creating_form.html [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+      <label for="course-group">Група</label>
+      <select id="course-group" name="group">
+        <option value="1">М</option>
+        <option value="2" selected >ПМ</option>
+        <option value="3">ОКН</option>
+        <option value="4">ЯКН</option>
+      </select>
+
+      <label for="course-credits">Кредити</label>
+      <input type="number "id="course-credits" name="credits">
```
Comment (martinely95) [inline Creating_form.html]: You could add here limit for minimum and maximum values. This way the user would need less attempts, clicks, thinking and time in order to fill the form correctly.

File: Form_validation.php [hunk 3]
```diff
@@ -0,0 +1,88 @@
+<?php 
+  $valid = array();
+  $errors = array();
+  if ($_POST) {
+
+    $title = $_POST['title'];
+
+    if (!$title) {
```
Comment (martinely95) [inline Form_validation.php]: You could exercise on extracting some functions here if you want.

File: Form_validation.php [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+        $errors['description'] = 'Описанието на дисциплината има минимална дължина 10 символа.';
+        echo $errors['description'];  
+      } 
+      else {
+        $valid['description'] = $description;   
+      }
+
+      $group = $_POST['group'];
+      if(!$group) {
+        //it is filled with the first option by default, so can not be missing
```
Comment (martinely95) [inline Form_validation.php]: In such case this is called "dead code" and you may want to delete it because it is unreachable. **BUT:** Also: never trust what comes from the fronted. For example I could use something like: https://www.postman.com/ and combine a POST query against the url where your backend is located by totally bypassing all your frontend validations. This is the reason why we always do validations both on frontend level and backend level as developers and we are basically "repeating" them - due to security reasons. The same goes for DB level. We usually have the same constraints that we have on the frontend and backend on the DB level too. This for example becomes useful when there are 2 types of backends connected to the same DB, and imagine the possibilities - the second backend may be missing some / all validations at this case the DB constraints come to play to protect the integrity of data. Also very often after an application that is in production there could happen some inconsistency on DB level (for example because of some bug) and the whole application to become unusable. In order to fix such situations developers need to manually execute SQL scripts against the DB. And in this case the DB level constraints are protecting us once again from ourselves and keeping data integrity.
**tl;dr;** please add the validation here as it actually CAN be missing.

File: Form_validation.php [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+      $credits = $_POST['credits'];
+      if($credits < 0) {
+        $errors['credits'] = 'Броят кредити трябва да е цяло положително число';
+        echo $errors['credits'];  
+      }
+      else{
+        $valid['credits'] = $credits;   
+      }
+	  
+	  if (count($valid) == 5) {
```
Comment (martinely95) [inline Form_validation.php]: What would happen if we add a new requirement and there is an error but the `$valid` count stays the same? Wouldn't it be less error-prone if we are checking for no errors? And also this way we will also eliminate this magical number `5`.

File: Form_validation.php [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+                  $filename = 'data.txt';
+                  file_put_contents($filename, "Име на предмет: ", FILE_APPEND | LOCK_EX);
```
Comment (martinely95) [inline Form_validation.php]: When I submit the form, I receive no result on the screen. Was everything successful? Did something fail? What exactly failed? What do I need to do to fix it? How can I (as a dumb user not knowing about the "back" button of the browser) navigate to the previous page and try to fill the form once again? What was the exact information that was recorded by the server to the file?
Here is the acceptance criteria for the task: https://iliasky.com/www/presentations/Forms.html?read=1#11

File: Form_validation.php [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+      else {
+        $valid['teacher'] = $teacher;   
+      }
+
+      $description = $_POST['description'];
+      if (!$description) {
+        $errors['description'] = 'Описанието на дисциплината е задължително поле.';
+        echo $errors['description'];   
+      } 
+      elseif (strlen($description) < 10)
```
Comment (martinely95) [inline Form_validation.php]: Please extract all constant values into constants/variables and give them meaningful names. This way you will know what is the purpose of the value when reading the code. For example here you would be able to reuse the value of 10 in the error message too here that way.
